src/Tools/ExternalAccess/Razor/Cohost/Handlers/Completion.cs

```diff
@@ -81,6 +81,11 @@ private static CompletionListCache GetCache()
         return InlineCompletionsHandler.GetInlineCompletionItemsAsync(logger, document, position, options, xmlSnippetParser, cancellationToken);
     }
 
+    public static Document ApplyProvisionalEdit(SourceGeneratedDocument generatedDocument, SourceText updatedDocumentText)
+    {
+        return generatedDocument.Project.Solution.WithFrozenSourceGeneratedDocument(generatedDocument.Identity, DateTime.Now, updatedDocumentText);
```

Comment:
@CyrusNajmabadi Did we use UtcNow for these, generally?

Reply:
Happy to update this, but just a note that the resulting document and the solution that contains it are thrown away immediately after asking for completion items from it

Reply:
Looks like DateTime.Now? https://github.com/dotnet/roslyn/blob/main/src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.RegularCompilationTracker_Generators.cs#L294

I might have missed a spot though, was hard enough tracking down that one :)